runtime/syntax/m4.vim

```diff
@@ -21,6 +21,7 @@ endif
 syn match  m4Variable contained "\$\d\+"
 syn match  m4Special  contained "$[@*#]"
 syn match  m4Comment  "\<\(m4_\)\=dnl\>.*" contains=SpellErrors
+syn match  m4Comment  "#.*" contains=SpellErrors
```

Comment:
SpellErrors is pretty old tech.  Should we update this to use the @Spell cluster?